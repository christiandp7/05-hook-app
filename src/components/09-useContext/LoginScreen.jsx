import React, { useContext } from 'react'
import { UserContext } from './UserContext';

const LoginScreen = () => {
  const { setUser } = useContext(UserContext)
  const newUser = {
    id: 123,
    name: 'Gianfranco',
    email: 'gianfranco@gmail.com'
  }
  return (
    <div>
      <h1>Login Screen</h1>
      <hr/>
      <button 
        className="btn btn-primary"
        onClick={() => setUser(newUser)}
      >Login</button>
    </div>
  )
}

export default LoginScreen
