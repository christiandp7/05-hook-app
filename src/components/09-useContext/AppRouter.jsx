import React from 'react'
import { BrowserRouter as Router, Route, Switch, Redirect } from 'react-router-dom'
// components
import AboutScreen from './AboutScreen'
import HomeScreen from './HomeScreen'
import LoginScreen from './LoginScreen'
import Navbar from './Navbar'


const AppRouter = () => {
  return (
    <Router>

        <Navbar />

        <div className="container">
          <Switch>
            <Route exact path="/" component={ HomeScreen } />
            <Route path="/about" component={ AboutScreen } />
            <Route path="/login" component={ LoginScreen } />
            <Route path="/404" component={HomeScreen} />
            <Redirect to="/404" />
          </Switch>
        </div>
        
    </Router>
  )
}

export default AppRouter
