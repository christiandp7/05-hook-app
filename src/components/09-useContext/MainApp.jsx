import React, { useState } from 'react'
import { Link } from 'react-router-dom'
import { UserContext } from './UserContext'

import AppRouter from './AppRouter'

const MainApp = () => {

  const [user, setUser] = useState({})

  return (
    <UserContext.Provider value={{
      user,
      setUser,
      temporal: 1234
    }}>
      <AppRouter />
    </UserContext.Provider>
  )
}

export default MainApp
