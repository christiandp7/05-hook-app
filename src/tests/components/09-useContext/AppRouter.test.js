import React from 'react'
import { mount } from 'enzyme'
import { UserContext } from '../../../components/09-useContext/UserContext'
import AppRouter from '../../../components/09-useContext/AppRouter'

describe('Test in <AppRouter />', () => {
	const user = {
		id: 1,
		name: 'Christian',
	}
	const wrapper = mount(
		<UserContext.Provider value={{ user }}>
			<AppRouter />
		</UserContext.Provider>,
	)

	test('should render correctly', () => {
		expect(wrapper).toMatchSnapshot()
	})
})
