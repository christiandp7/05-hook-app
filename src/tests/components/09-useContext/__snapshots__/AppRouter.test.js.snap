// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <AppRouter /> should render correctly 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <div
      className="container-fluid"
    >
      <a
        aria-current="page"
        className="navbar-brand active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        UseContext
      </a>
      <div
        className="collapse navbar-collapse"
        id="navbarNav"
      >
        <ul
          className="navbar-nav"
        >
          <li
            className="nav-item"
          >
            <a
              aria-current="page"
              className="nav-link active"
              href="/"
              onClick={[Function]}
              style={Object {}}
            >
              Home
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              aria-current={null}
              className="nav-link"
              href="/about"
              onClick={[Function]}
            >
              About
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              aria-current={null}
              className="nav-link"
              href="/login"
              onClick={[Function]}
            >
              Login
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>,
  <div
    className="container"
  >
    <div>
      <h1>
        Home Screen
      </h1>
      <hr />
      <pre>
        {"id":1,"name":"Christian"}
      </pre>
    </div>
  </div>,
]
`;
